### PR TITLE
Pcsx2Types fix for visual studio

### PR DESCRIPTION
**Pull request by uyjulian**
Needed before #476

### CODE REVIEW

**Comment by avih** [comment]
Some more info at the commit comment could be useful. Are these `typedef`s not required anymore for MSVC++ 11 onwards?

**Comment by uyjulian** [comment]
no, newer version of visual studio has `stdint.h`

**Comment by avih** [comment]
But the patch doesn't add `#include <stdint.h>` for MSVC versions which have it ?

**Comment by uyjulian** [comment]
it defines `uintptr_t`/`intptr_t` for those MSVC that don't include `stdint.h`
otherwise, error occur

File: common/include/Pcsx2Types.h [hunk 1]
```diff
@@ -56,6 +59,7 @@ typedef unsigned __int64 u64;
 // char and not signed char in pcsx2, we define s8 to char
 
 typedef char s8;
+typedef int8_t s8_proper;
```
Comment (avih) [inline common/include/Pcsx2Types.h]: What's `s8_proper` and why did you add this line?

Comment (uyjulian) [inline common/include/Pcsx2Types.h]: Oh, whoops, forgot to remove that.

**Comment by avih** [comment]
Please also update the commit message to explain what is and what isn't required on different versions of MSVC, and then I'll merge it. Thanks.

**Comment by avih** [comment]
And just to make sure, did you verify that it works on VS2012 ans VS 2013?

**Comment by uyjulian** [comment]
Yes, I verified it works on all VS2012/VS2013

I fixed the diff to be simpler.

**Comment by avih** [comment]
After the change - now I completely don't get what is it good for. `uptr` and `sptr` were already defined before this PR outside any `#ifdef`, what good does it make defining them differently for MSVC? and where the `_MSC_VER` vresion threshold go?

Could you please explain what this PR tries to fix (both the original one or the new one)?

**Comment by uyjulian** [comment]
Okay, the new PR attempts to fix the error when `stdint.h` is not included, so I stuck line 72/73 inside the `#ifdef`/`#else` block, because `uintptr_t`/`intptr_t` relies on `#include "stdint.h"`.

**Comment by avih** [comment]
What error when `stdint.h` is not included? I can compile PCSX2 and I see no error...

Again, from the beginning:
1. What problem does this patch try to fix?
2. On what cases does the problem occure? (E.g. I don't think I can see this problem).
3. Why do you define `uptr` and `sptr` to be `u32` and `s32` and not `int*` and `unsigned int*` ?

Please, don't be cryptic in your PRs and replies. When you make a PR, I need to be able to understand everything by reading the commit message. Not discuss it with you over 10 comments.

**Comment by uyjulian** [comment]
I wonder why I opened this PR in the first place, I'm really confused now... 
1. This patch tries to fix #476 not compiling because `uintptr_t`/`intptr_t` isn't defined for some reason, which causes an error... (And I have no idea why)
2. On #476 the problem occurs
3.

**Comment by avih** [comment]
I'll close this PR then. If/when you figure out exactly what you want to fix and how the patch fixes it, post another comment here and I'll reopen it.
